WowUp.WPF/App.xaml.cs

```diff
@@ -61,14 +69,67 @@ public App()
 
             _serviceProvider = serviceCollection.BuildServiceProvider();
             _analyticsService = _serviceProvider.GetRequiredService<IAnalyticsService>();
+            _sessionService = _serviceProvider.GetRequiredService<ISessionService>();
+            _addonService = _serviceProvider.GetRequiredService<IAddonService>();
         }
 
+        async void TestParsed(StartupOptions options)
+        {
+            if (options.ClientType != Common.Enums.WowClientType.None)
```

Comment:
I think a lot of this could be removed/moved. For something like start minimized you could check the args in the mainwindow and just set the mode to minimized.

Reply:
I see. However, I thought of handling it with keeping in mind that only one instance of the app can run at a time. I agree though that it is can be ignored for all options except for --install. Considering that it is still necessary to pass args between app instances in this case, can you advice how to do that? I can use MarshalByRefObject or dig into the IPC and see how it can be done with it.

Reply:
in the case of multiple instances and command line i would probably just throw some error and say close other instances before trying to run the command line again

Reply:
For most of them, yes. I didn't add any errors though because of handling startup args in defferent places in code. I also created a static class to store parse result and avoid calling Parse multiple times.

There is, however, an exception - I think --install should work even if an instance is already running. This is kind of required to make my brwoser extension concept work. Didn't implement this behavior yet.

In any case, I am running out of time right now. The last commit seems like it can pretty much be pulled. Please review it.